tutorials/common/common_vaapi.cpp

```diff
@@ -512,12 +512,19 @@ mfxStatus _simple_free(mfxHDL pthis, mfxFrameAllocResponse* response)
     mfxU32 i = 0;
     bool isBitstreamMemory = false;
     bool actualFreeMemory = false;
-    if (0 == memcmp(response, &(allocDecodeResponses[pthis].mfxResponse),
-                    sizeof(*response))) {
-        allocDecodeResponses[pthis].refCount--;
-        if (0 == allocDecodeResponses[pthis].refCount)
-            actualFreeMemory = true;
-    }; // else actualFreeMemory = true;
+
+    if (--allocDecodeResponses[pthis].refCount == 0) {
+        // Decode free response handling
+        if (0 == memcmp(response, &(allocDecodeResponses[pthis].mfxResponse),
+                        sizeof(*response))) {
+            allocDecodeResponses[pthis].refCount--;
```

Comment:
You have already decremented this counter to zero at line 516

Reply:
@mgonchar, fixed